lib/ui/painting.dart

```diff
@@ -1316,6 +1316,26 @@ Future<Null> _decodeImageFromListAsync(Uint8List list, ImageDecoderCallback call
   callback(frameInfo.image);
 }
 
+/// The encoding formats supported by the [encodeImage].
+enum EncodedImageFormat {
+  // Be conservative with the formats we expose. We can increase the list in future but reducing
+  // it is more difficult.
+  JPEG,
+  PNG,
+  WEBP,
+}
+
+/// Callback signature for [encodeImage].
+typedef void ImageEncoderCallback(Uint8List result);
```

Comment:
Please document what the "result" argument is (presumably, bytes representing the encoded image in the requested format?).

Reply:
(this can just be removed if we go the route suggested below of returning a future)

Reply:
Removed. Using a Future instead.